pkg/clients/eks/nodegroup.go

```diff
@@ -78,6 +86,20 @@ func GenerateUpdateNodeGroupConfigInput(name string, p *v1alpha1.NodeGroupParame
 			MinSize:     p.ScalingConfig.MinSize,
 			MaxSize:     p.ScalingConfig.MaxSize,
 		}
+
+		// If desiredSize is not set, derive the value from either the
+		// current observed desiredSize, or the min/max if observed is out of bounds.
+		if p.ScalingConfig.DesiredSize == nil {
+			// The min/max size set the floor/ceiling for the desiredSize
+			switch desiredSizeVal := awsclients.Int64Value(ng.ScalingConfig.DesiredSize); {
```

Comment:
I wouldn't really do those checks because we should aim for making these logic functions as dumb as possible. So, if desired size is to be ignored, we should just use the observed size.

Reply:
That was my initial implementation. When testing I ran into some issues where the crd had say `minSize` of 1 and `maxSize` of 10, and the nodeGroup had been scaled by another entity to a desired size of 4. Because of the value of the desiredSize, increasing the `minSize` of the nodeGroup  > 4 will return errors from AWS as this would cause desired size to be less than `minSize`.

My thinking is that because the user has provided a min/max size without a desired size, when creating we can derive a suitable desired size using the min size as an initial value.. When updating we can derive one as well, either the current observed value for desired size, or the bounds of the min/max size.